File: JavaScript/DailyCodeChallenge/260123_isValidHex.js
function isValidHex(str) {

  const regex = /#([0-9a-f]{6}|[0-9a-f]{3})$/i

  console.log(regex.test(str))

  return regex.test(str);
}

isValidHex("#1234567")

/*
Hex Validator
Given a string, determine whether it is a valid CSS hex color. A valid CSS hex color must:

Start with a #, and
be followed by either 3 or 6 hexadecimal characters.
Hexadecimal characters are numbers 0 through 9 and letters a through f (case-insensitive).

Tests
Passed:1. isValidHex("#123") should return true.
Passed:2. isValidHex("#123abc") should return true.
Passed:3. isValidHex("#ABCDEF") should return true.
Passed:4. isValidHex("#0a1B2c") should return true.
Passed:5. isValidHex("#12G") should return false.
Passed:6. isValidHex("#1234567") should return false.
Passed:7. isValidHex("#12 3") should return false.
Passed:8. isValidHex("fff") should return false.

*/

File: Python/Daily_Code_Challenge/260123-is-valid-hex.py
import re

def is_valid_hex(s):

    regex = r'#([0-9a-f]{6}|[0-9a-f]{3})$'

    string = re.search(regex,s,flags=re.IGNORECASE)

    print(bool(string))

    return bool(string)

is_valid_hex("#123")
is_valid_hex("#1234567")

'''
Hex Validator
Given a string, determine whether it is a valid CSS hex color. A valid CSS hex color must:

Start with a #, and
be followed by either 3 or 6 hexadecimal characters.
Hexadecimal characters are numbers 0 through 9 and letters a through f (case-insensitive).

Tests
Passed:1. is_valid_hex("#123") should return True.
Passed:2. is_valid_hex("#123abc") should return True.
Passed:3. is_valid_hex("#ABCDEF") should return True.
Passed:4. is_valid_hex("#0a1B2c") should return True.
Passed:5. is_valid_hex("#12G") should return False.
Passed:6. is_valid_hex("#1234567") should return False.
Passed:7. is_valid_hex("#12 3") should return False.
Passed:8. is_valid_hex("fff") should return False.
'''